source/common/statistic_impl.cc

```diff
@@ -236,4 +236,103 @@ nighthawk::client::Statistic HdrStatistic::toProto(SerializationDomain domain) c
   return proto;
 }
 
-} // namespace Nighthawk
\ No newline at end of file
+CircllhistStatistic::CircllhistStatistic() {
+  histogram_ = hist_alloc();
+  ASSERT(histogram_ != nullptr);
+}
+
+CircllhistStatistic::~CircllhistStatistic() { hist_free(histogram_); }
+
+void CircllhistStatistic::addValue(uint64_t value) {
+  hist_insert_intscale(histogram_, value, 0, 1);
+  StatisticImpl::addValue(value);
+}
+double CircllhistStatistic::mean() const { return hist_approx_mean(histogram_); }
+double CircllhistStatistic::pvariance() const { return pstdev() * pstdev(); }
+double CircllhistStatistic::pstdev() const {
+  return count() == 0 ? std::nan("") : hist_approx_stddev(histogram_);
+}
+
+StatisticPtr CircllhistStatistic::combine(const Statistic& statistic) const {
+  auto combined = std::make_unique<CircllhistStatistic>();
+  const auto& stat = dynamic_cast<const CircllhistStatistic&>(statistic);
+  hist_accumulate(combined->histogram_, &this->histogram_, /*cnt=*/1);
+  hist_accumulate(combined->histogram_, &stat.histogram_, /*cnt=*/1);
+
+  combined->min_ = std::min(this->min(), stat.min());
+  combined->max_ = std::max(this->max(), stat.max());
+  combined->count_ = this->count() + stat.count();
+  return combined;
+}
+
+StatisticPtr CircllhistStatistic::createNewInstanceOfSameType() const {
+  return std::make_unique<CircllhistStatistic>();
+}
+
+nighthawk::client::Statistic CircllhistStatistic::toProto(SerializationDomain domain) const {
+  nighthawk::client::Statistic proto = StatisticImpl::toProto(domain);
+  if (count() == 0) {
+    return proto;
+  }
+
+  // List of quantiles is based on hdr_proto_json.gold which lists the quantiles provided by
```

Comment:
Note: I think HdrHistogram will emit more percentiles as the number of (differently-valued) data points added grows. So it's not a fixed set with HdrHistogram. 
I am not sure if there is an equivalent way to iterate the available buckets with libcircllhist to get more detail out, and I am also not sure this is really a problem, but I thought it would be good to call out the difference here. 

(The only direct implication I figure it would have today is that the Fortio plots would be drawn with less data points)

Reply:
Thanks Otto for the information! Updated the comment to just `List of quantiles is based on hdr_proto_json.gold.`
From the APIs in [circllhist.h](https://github.com/circonus-labs/libcircllhist/blob/master/src/circllhist.h), there is no easy way to iterate the quantiles. If the listed data points turn out to be not enough in the future, we can always add more here.